load_test/lib/load_test/user/sse.ex

```diff
@@ -59,10 +69,18 @@ defmodule SseUser do
       }
     }
 
-    {:ok, conn_pid} = :gun.open(String.to_atom(parsed_url.host), parsed_url.port, opts)
-    {:ok, proto} = :gun.await_up(conn_pid)
-    Logger.debug(fn -> "Connection established with proto #{inspect(proto)}" end)
-    stream_ref = :gun.get(conn_pid, parsed_url.path, headers)
+    connect_callback = fn state ->
```

Comment:
**Suggestion** 🤔 : maybe this would be better in a defp (like reconnect), just to have a clear view of the inputs, then use the &function_name/arity syntax when passing you need to pass it

Reply:
the issue here is this subfunction require a lot of parameters